tsnet/tsnet_test.go

```diff
@@ -167,3 +173,59 @@ func TestConn(t *testing.T) {
 		t.Errorf("got %q, want %q", got, want)
 	}
 }
+
+func TestLocalAPI(t *testing.T) {
+	controlURL := startControl(t)
+
+	tmp := t.TempDir()
+	tmps1 := filepath.Join(tmp, "s1")
+	os.MkdirAll(tmps1, 0755)
+	s1 := &Server{
+		Dir:        tmps1,
+		ControlURL: controlURL,
+		Hostname:   "s1",
+		Store:      new(mem.Store),
+		Ephemeral:  true,
+	}
+	defer s1.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	if _, err := s1.Up(ctx); err != nil {
+		t.Fatal(err)
+	}
+
+	addr, cred, err := s1.LocalAPI()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	url := "http://" + addr + "/localapi/v0/status"
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	res.Body.Close()
+	if res.StatusCode != 403 {
+		t.Errorf("GET %s returned %d, want 403 permission denied without header", url, res.StatusCode)
+	}
+
+	req, err = http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set("Sec-Tailscale-LocalAPI", cred)
```

Comment:
Can we please also have a test with the header but an incorrect value for it?

Reply:
I switched the cred over to the RequiredPassword machinery that's tested elsewhere, so I believe "" and the correct value are enough now.